proto/spacemesh/api.proto

```diff
@@ -1,166 +1,70 @@
 
-message SyncStatus {
-  enum SyncStatus {
-      UNDEFINED = 0;
-      NEW_LAYER_VERIFIED = 1;
-      NEW_TOP_LAYER = 2;
-      SYNCED = 3;
-      SYNCING = 4;
-      NOT_SYNCED = 5;
-  }
-}
+    // Returns number of layers per epoch (a network constant)
+    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
 
-service SpacemeshService {
-    // SCALAR
+    ////////// Node streaming data facade
 
-    // Test facade
+    // Sync status events
+    rpc NodeSyncStatuses(google.protobuf.Empty) returns (stream NodeSyncStatus);
 
-    // A simple test endpoint
-    rpc Echo(SimpleMessage) returns (SimpleMessage);
+    // Node error events
+    rpc NodeErrors(google.protobuf.Empty) returns (stream NodeError);
 
-    // Node facade
+    //////////////// Mesh streaming data facade
 
-    // Returns the version of the node software as a semver string
-    rpc NodeVersion(google.protobuf.Empty) returns (SimpleMessage);
-    // Returns node status object
-    rpc NodeStatus(google.protobuf.Empty) returns (NodeStatus);
-    // Tell the node to start sync
-    rpc NodeSyncStart(google.protobuf.Empty) returns (google.protobuf.Empty);
+    // Receipts - emitted after tx was processed by STF (or rejected before STF)
+    rpc MeshTransactionReceipts(google.protobuf.Empty) returns (stream TransactionReceipt);
 
-    // Mesh facade
+    // App Events - emitted by app methods impl code trigged by an app transaction
+    rpc MeshAppEvents(google.protobuf.Empty) returns (stream AppEvent);
 
-    // Returns mesh genesis time (as UTC string)
-    rpc MeshGenesisTime(google.protobuf.Empty) returns (SimpleMessage);
-    // Returns current layer
-    rpc MeshCurrentLayer(google.protobuf.Empty) returns (SimpleInt);
-    // Returns current epoch
-    rpc MeshCurrentEpoch(google.protobuf.Empty) returns (SimpleInt);
-    // Returns network ID
-    rpc MeshNetId(google.protobuf.Empty) returns (SimpleInt);
-    // Returns number of layers per epoch
-    rpc MeshEpochNumLayers(google.protobuf.Empty) returns (SimpleInt);
+    // Accounts
+    rpc MeshAccounts(google.protobuf.Empty) returns (stream Account);
 
-    // STREAM
+    // Rewards
+    rpc MeshRewards(google.protobuf.Empty) returns (stream Reward);
 
-    // Node stream facade
+    // Layers
+    rpc MeshLayer(google.protobuf.Empty) returns (stream Layer);
 
-    // Sync status events
-    rpc NodeStreamSyncStatus(google.protobuf.Empty) returns (stream SyncStatus);
-    // Node error events
-    rpc NodeStreamError(google.protobuf.Empty) returns (stream NodeError);
-
-    // Mesh stream facade
-
-    // Account events
-    rpc MeshStreamAccount(google.protobuf.Empty) returns (stream AccountEvent);
-    // Reward events
-    rpc MeshStreamReward(google.protobuf.Empty) returns (stream Reward);
-    // Layer events
-    rpc MeshStreamLayer(google.protobuf.Empty) returns (stream Layer);
-    // Layer state change events
-    rpc MeshStreamLayerState(google.protobuf.Empty) returns (stream SimpleInt);
```

Comment:
What happened to this stream?

Reply:
I don't think we need it. I think it is good design to just have stream of entities - layers, blocks, transactions and just have the state of these data objects up-to-date so it is the client responsibility to update its stored data based on the last entity it got on a stream and can maintain state like that. So, the idea is that instead sending a this-object-changed-state in a method - we just send the object on its standard stream with the updated state and the client inspects it and updates itself... I think this is a good pattern - I know it was me who suggested this method before...

Reply:
Resolved in #5